source/1_engine/rex_engine/include/rex_engine/core_application.h

```diff
@@ -60,6 +63,7 @@ namespace rex
     void loop();
 
   private:
+    bool m_is_paused;
```

Comment:
I think it's better to have an enum for the state of the application instead of adding flags.
in the future we might have more complex states